spec/draft/API_specification/indexing.rst

```diff
@@ -173,6 +179,54 @@ Multi-dimensional arrays must extend the concept of single-axis indexing to mult
 
     *Rationale: this is consistent with bounds-checking for single-axis indexing. An implementation may choose to set the axis (dimension) size of the result array to* ``0`` *, raise an exception, return junk values, or some other behavior depending on device requirements and performance considerations.*
 
+Integer Array Indexing
+----------------------
+
+An array must support indexing a one-dimensional array by an integer array (or an equivalent sequence) according to the following rules. Let ``A`` be a one-dimensional array with shape ``(n,)``, and let ``J`` be an integer array (or an equivalent sequence).
```

Comment:
Nit: I found it a little confusing that this section starts by describing special behavior for 1D ararys, which is entirely a special case of the behavior for ND arrays. It would be helpful to clarify this up front.

(The only way in which 1D arrays need to be special cased is that indexing like `a[b]` should be defined as equivalent to  `a[(b,)]`, but that is almost just a detail of Python syntax.)

Reply:
I've removed and reworked as part of generalization to ND arrays.